### PR TITLE
General cleanup pass with calculator/feed refactoring

### PR DESCRIPTION
**Pull request by mcsloy**
- `Calculator.__call__` method arguments `geometry` and `orbs` mandatory, and removed associated existence dependent code.
- `Calculator.__call__` method forwards `geometry` and `orbs` arguments to the `Calculator.forwards` method.
- `Calculator.__call__` method now performs a reset unconditionally.
- Moved `Calculator.__call__` functionality over to the `Calculator.forwards` method.
- Removed `Calculator.__call__` method overrides to better align them with PyTorch standards. Redirected `Calculator.forward` calls to the `Calculator.__call__` method as inherited from `torch.nn.Module`.
- Repeated `__call__` method removal for the `Feed` classes.
- Removed unused attribute "`_ml_params`" from the `Calculator` class.
- Added in special catch to the `Skf` file parser to handle unique deviations from the skf file format standard present in the pbc parameter set.
- Modified `Skf.blocks` method to reduce dependency on the `Skf.on_site` attribute's type. This should make generalising the on-sites entity to different types later on (e.g. `Feed` entities).

### CODE REVIEW

**Comment by WbSun723** [comment]
Hey Adam, all looks fine to me apart from a minor comment I made before, nice work!
BTW, I just made a simple test of the io codes for the pbc case, yes now all the zero components can be masked out in the hdf5 files. Also, now if the definition of basis in 'OrbitalInfo' doen't match with sk files, an IndexError will occur. But I think this is not the case for the previous version: because in the unittest for pbc set, Si is defined as s,p basis but actually d oribital is still included in the sk files. But unittest can pass and there is no IndexError shown.

File: tbmalt/common/maths/interpolation.py
[the remark this answers is not in the document]
Comment (mcsloy) [reply tbmalt/common/maths/interpolation.py:1]: Good catch, I have updated the feature branch so that the examples import the necessary classes.

**Comment by mcsloy** [comment]
> Hey Adam, all looks fine to me apart from a minor comment I made before, nice work! BTW, I just made a simple test of the io codes for the pbc case, yes now all the zero components can be masked out in the hdf5 files. Also, now if the definition of basis in 'OrbitalInfo' doen't match with sk files, an IndexError will occur. But I think this is not the case for the previous version: because in the unittest for pbc set, Si is defined as s,p basis but actually d oribital is still included in the sk files. But unittest can pass and there is no IndexError shown.

I have modified the `SkFeed` class so it only interacts with the on-site object when retrieving data from it, which helps to avoid unintentional dependencies on the on-site entity's type. This will make it easier to extend the `SkFeed` class so that users may supply on-sites as `Feed` instances, not just as dictionaries.

Previously, `SkFeed` instances would not necessarily error out if the `OrbitalInfo` and `on_site` instances disagreed on the number of orbitals. However, in practice I would view two parts of a model disagreeing on the number of orbitals to be an error.

That being said, as discussed in the last project meeting, we do need to take care to ensure that skf files are read as expected to avoid any potential issues this change might cause.
